py-polars/tests/test_cfg.py

```diff
@@ -66,23 +66,92 @@ def test_tbl_width_chars(environ: None) -> None:
     assert max(len(line) for line in str(df).split("\n")) == 13
 
 
-def test_tbl_cols(environ: None) -> None:
+def test_set_tbl_cols(environ: None) -> None:
     df = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9]})
 
     pl.Config.set_tbl_cols(1)
     assert str(df).split("\n")[2] == "│ a   ┆ ... │"
     pl.Config.set_tbl_cols(2)
-    assert str(df).split("\n")[2] == "│ a   ┆ ... ┆ c   │"
+    assert str(df).split("\n")[2] == "│ a   ┆ b   ┆ c   │"
     pl.Config.set_tbl_cols(3)
     assert str(df).split("\n")[2] == "│ a   ┆ b   ┆ c   │"
 
+    df = pl.DataFrame(
+        {"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9], "d": [10, 11, 12]}
+    )
+    pl.Config.set_tbl_cols(2)
+    assert str(df).split("\n")[2] == "│ a   ┆ ... ┆ d   │"
+    pl.Config.set_tbl_cols(3)
+    assert str(df).split("\n")[2] == "│ a   ┆ b   ┆ c   ┆ d   │"
```

Comment:
This seems like an off-by-one error: if three cols are requested, why does it return four?

Reply:
fixed